heltour/tournament/models.py

```diff
@@ -1872,6 +1879,15 @@ def seed_rating_display(self, league=None):
             if league is None:
                 league = self.season.league
             return self.player.rating_for(league)
+    
+    def withdraw_from_team_season(self, round):
```

Comment:
I like the idea of making this be an `@classmethod` so it can be called like I suggested in the other comment, which makes this signature read:
```python
@classmethod
def withdraw_from_season(cls, round, player):
    if not TeamMember.objects.filter(player=self.player, team__season=round.season).exists():
        cls.objects.filter(season=self.round.season, player=self.player).update(is_active=False)
        self.is_active = False
        self.save()
    cls._set_unavailable_for_season(player, skip_current=True)
    add_system_comment(self, 'player withdrawn: %s'%round)
```

Note: this makes it so that you have to update `_set_unavailable_for_season` to be callable either as a `@classmethod` or as an instance method. Which shouldn't be too hard. Maybe just make a new one that's the `@classmethod` with a different name and update the current one to use it.

Reply:
as we need the `SeasonPlayer` for add_system_comment as well, I am getting the correct `SeasonPlayer sp` and then call `sp._set_unavailable_for_season` instead of making another `@classmethod`